src/Exception/ApiResponseException.php

```diff
@@ -119,11 +119,17 @@ private function parseErrorResponse(ResponseInterface $response): array
                     if (array_key_exists('code', $array)) {
                         $error['code'] = $array['code'];
                     }
+                    elseif (isset($array['error']['code'])) {
+                        $error['code'] = $array['error']['code'];
+                    }
                     if (array_key_exists('message', $array)) {
                         // It could happen that the returned message by
                         // Apigee Edge is also an array.
                         $error['message'] = is_array($array['message']) ? json_encode($array['message']) : $array['message'];
                     }
+                    elseif (isset($array['error']['message'])) {
+                        $error['message'] = $array['error']['message'];
```

Comment:
The status information in the message looks also important. Should not it be concatenated with the message string if it is set in the API response, so when this exception gets logged, the log entry would contain all information?

> "status": "PERMISSION_DENIED"

Reply:
@mxr576: I've added the status message if available.